src/api/java/com/minecolonies/api/tileentities/TileEntityRack.java

```diff
@@ -447,13 +445,13 @@ public void rotate(final Rotation rotationIn)
     @Override
     public <T> LazyOptional<T> getCapability(@Nonnull final Capability<T> capability, final Direction dir)
     {
-        if (lastOptional != null)
-        {
-            lastOptional.invalidate();
-        }
-
         if (capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY)
         {
+            if (lastOptional != null && lastOptional.isPresent())
+            {
+                return (LazyOptional<T>) lastOptional;
```

Comment:
Ehm, shouldn't we avoid returning this cached value here?

Reply:
No why?

Reply:
I have a feeling I'm talking to a wall. I discussed this 2 weeks ago and removed this caching stuff.

Reply:
Because you think it will break understood.
But I have no evidence that it will. It works for tons of other mods, even for vanilla. So come up with evidence that it will break.

He invalidates the cache when the single or double status changes. Which is the only relevant thing, in my eyes.

Reply:
No no, not I think it will break, it did break in numerous cases and items were lost, not detected by dman, duped, etc

Reply:
I'm talking with someaddons about this and forge does it completely different for chests too, (no simple cache like this) so we should probably copy this behaviour

Reply:
If that is the case, then go for it. I would follow the example of the chests and just stick with that.
I thought they cached it that is why I confused. If they do not then disregard my opinion.

Reply:
their lazyoptional basically queries the tileEntity from the world =D